src/sinks/elasticsearch/mod.rs

```diff
@@ -189,14 +190,28 @@ impl fmt::Display for VersionValueParseError<'_> {
 impl ElasticsearchCommonMode {
     fn index(&self, log: &LogEvent) -> Option<String> {
         match self {
-            Self::Bulk { index, .. } => index
+            Self::Bulk {
+                index,
+                template_fallback_index,
+                ..
+            } => index
                 .render_string(log)
-                .map_err(|error| {
-                    emit!(TemplateRenderingError {
-                        error,
-                        field: Some("index"),
-                        drop_event: true,
-                    });
+                .or_else(|error| {
+                    if let Some(fallback) = template_fallback_index {
+                        emit!(TemplateRenderingError {
```

Comment:
We typically only emit error level events in sinks when they are non-recoverable. Here it is falling back to the configured default index. I think we can just drop this emit since if users are configuring a fallback index they are expecting some events to fail and so there isn't really anything to alert them to.

I could see some users wanting a metric about how many events failed to render the index, but likely we don't need a log. I'll leave it up to you if you want to add a metric. We could wait for someone to ask for it.

Reply:
Intended use-case for this is to ensure logs are not dropped, but also inform the user if a field they are expecting to be present in a log is not there. Wanted to keep this `emit` to make sure the user is kept informed in these scenarios

Reply:
Gotcha, I see. I didn't realize that `TemplateRenderingError` is already emitting as a warning if events aren't dropped so 👍  to rolling with this.